### PR TITLE
[SW-1391] controller that lets you specify position, velocity, and load at the same time 

### PR DESCRIPTION
**Pull request by khughes-bdai**
## Change Overview

Adds a new package `spot_controllers` that can contain custom ros2 controllers for spot. 

Adds a custom controller `forward_state_controller`. This uses the same base class that the built-in `forward_command_controller` and `multi_interface_forward_command_controller` use. The main difference is that this controller accepts a list of joints and a list of interfaces to forward commands over.  This controller is then set up with `spot_ros2_control` so that you can forward a list of position, velocity, and load commands to the low level API at the same time. 

To launch with this controller add the launch argument `robot_controller:=forward_state_controller`

To send commands, publish to the `/<robot name>/forward_state_controller/commands` topic, eg
 
`ros2 topic pub /forward_state_controller/commands std_msgs/msg/Float64MultiArray "{data: [<list of desired position> <list of desired velocity> <list of desired load>]}"
`

Additionally, if joint gains are ever added to the command interface, they could be forwarded via this same type of controller by just inputting a different set of parameters

## Testing Done

- mock mode / forward position controller
-  mock mode / forward state controller
-  robot mode / forward position controller
- robot mode / forward state controller
all are loaded successfully!
- [x] mock mode correctly forwards a dummy list of position/velocity/effort into the joint states
- [x] examples running regular forward position commands still work

### CODE REVIEW

**Comment by khughes-bdai** [comment]
* **#493** <a href="https://app.graphite.dev/github/pr/bdaiinstitute/spot_ros2/493?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @khughes-bdai and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: spot_controllers/include/spot_controllers/forward_state_controller.hpp [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+ *
+ * This class forwards the command signal for a set of interfaces over a set of joints.
+ *
+ * \param joints Names of the joint to control.
+ * \param interface_names Names of the interfaces to command.
+ *
+ * Subscribes to:
+ * - \b commands (std_msgs::msg::Float64MultiArray) : The commands to apply.
+ */
+class ForwardStateController : public forward_command_controller::ForwardControllersBase {
```
Comment (khughes-bdai) [inline spot_controllers/include/spot_controllers/forward_state_controller.hpp:41]: open to suggestions of potential better names than ForwardStateController. The main idea is just that it takes in a list of joints and a list of interfaces. 
Builtin controllers already have ForwardCommandController (list of joints, one interface) and MultiInterfaceForwardCommandController (one joint, list of interfaces) https://github.com/ros-controls/ros2_controllers/blob/humble/forward_command_controller/forward_command_plugin.xml

Comment (tcappellari-bdai) [reply spot_controllers/include/spot_controllers/forward_state_controller.hpp:41]: I think ForwardStateController is fine

**Review by tcappellari-bdai — APPROVED** [review]
lgtm :tada:

File: spot_controllers/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+
+# Add the hardware interface
+add_library(
+  spot_controllers
+  SHARED
+  src/forward_state_controller.cpp
+)
+target_compile_features(spot_controllers PUBLIC cxx_std_20)
+target_include_directories(spot_controllers PUBLIC
+$<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/include>
```
Comment (mhidalgo-bdai) [inline spot_controllers/CMakeLists.txt:48]: @khughes-bdai nit: can we format this a bit?

**Review by mhidalgo-bdai — APPROVED** [review]
Looks correct. Trust your testing here @khughes-bdai.

**Comment by khughes-bdai** [comment]
### Merge activity

* **Oct 3, 10:22 AM EDT**: @khughes-bdai started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/bdaiinstitute/spot_ros2/493).

**Review by tcappellari-bdai — APPROVED** [review]

**Pull request merged**